test/framework/workload_flux.go

```diff
@@ -0,0 +1,98 @@
+package framework
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	"github.com/aws/eks-anywhere/pkg/constants"
+	"github.com/aws/eks-anywhere/pkg/filewriter"
+	"github.com/aws/eks-anywhere/pkg/git"
+)
+
+func (w *WorkloadCluster) clusterConfGitPathInManagementCluster() string {
+	p := w.ManagementCluster.ClusterConfig.FluxConfig.Spec.ClusterConfigPath
+	if len(p) == 0 {
+		p = path.Join("clusters", w.managementCluster().Name)
+	}
+
+	return path.Join(p, w.ClusterName, constants.EksaSystemNamespace, eksaConfigFileName)
+}
+
+func (w *WorkloadCluster) clusterConfigGitPathInManagementCluster() string {
+	return filepath.Join(w.ManagementCluster.GitWriter.Dir(), w.clusterConfGitPathInManagementCluster())
+}
+
+func (w *WorkloadCluster) addConfigChangesInManagementClusterToGit(ctx context.Context) error {
+	p := w.clusterConfGitPathInManagementCluster()
+	g := w.ManagementCluster.GitClient
+	if err := g.Add(p); err != nil {
+		return fmt.Errorf("adding cluster path changes at %s: %v", p, err)
+	}
+
+	if err := w.pushConfigChangesInManagementCluster(ctx, "EKS-A E2E Flux test configuration changes added"); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (w *WorkloadCluster) removeConfigInManagementClusterFromGit(ctx context.Context) error {
+	p := w.clusterConfGitPathInManagementCluster()
+	g := w.ManagementCluster.GitClient
+	if err := g.Remove(p); err != nil {
+		return fmt.Errorf("removing cluster config at path %s: %v", p, err)
+	}
+
+	if err := w.pushConfigChangesInManagementCluster(ctx, "EKS-A E2E Flux test configuration removed"); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (w *WorkloadCluster) pushConfigChangesInManagementCluster(ctx context.Context, commitMessage string) error {
+	g := w.ManagementCluster.GitClient
```

Comment:
same comment about accessing the management cluster. all this code suffers from deep coupling and unclear responsibility boundary. Let's rethink this.